react/features/file-sharing/middleware.web.ts

```diff
@@ -23,22 +26,66 @@ import { downloadFile } from './utils';
  */
 StateListenerRegistry.register(
     state => state['features/base/conference'].conference,
-    (conference, { dispatch }, previousConference) => {
+    (conference, { dispatch, getState }, previousConference) => {
         if (conference && !previousConference) {
             conference.on(JitsiConferenceEvents.FILE_SHARING_FILE_ADDED, (file: IFileMetadata) => {
                 dispatch(addFile(file));
+
+                const state = getState();
+                const localParticipant = getLocalParticipant(state);
+                const isRemoteFile = file.authorParticipantId !== localParticipant?.id;
+
+                if (isRemoteFile) {
+                    const { isOpen, focusedTab } = state['features/chat'];
+                    const isFileSharingTabVisible = isOpen && focusedTab === ChatTabs.FILE_SHARING;
+
+                    if (!isFileSharingTabVisible) {
+                        dispatch({
+                            type: INCREMENT_UNREAD_FILES_COUNT
+                        });
+                    }
+
+                    dispatch(showNotification({
+                        titleKey: 'fileSharing.newFileNotification',
+                        titleArguments: { participantName: file.authorParticipantName, fileName: file.fileName }
+                    }, NOTIFICATION_TIMEOUT_TYPE.SHORT));
+                }
             });
             conference.on(JitsiConferenceEvents.FILE_SHARING_FILE_REMOVED, (fileId: string) => {
+                const state = getState();
+                const localParticipant = getLocalParticipant(state);
+                const { files } = state['features/file-sharing'];
+                const removedFile = files.get(fileId);
+
+                if (removedFile && removedFile.authorParticipantId === localParticipant?.id) {
+                    dispatch(showNotification({
+                        titleKey: 'fileSharing.fileRemovedByOther',
+                        titleArguments: { fileName: removedFile.fileName },
+                        appearance: NOTIFICATION_TYPE.WARNING
+                    }, NOTIFICATION_TIMEOUT_TYPE.MEDIUM));
+                }
+
                 dispatch({
                     type: _FILE_REMOVED,
                     fileId
                 });
             });
 
             conference.on(JitsiConferenceEvents.FILE_SHARING_FILES_RECEIVED, (files: object) => {
+                const state = getState();
+                const localParticipant = getLocalParticipant(state);
+                let remoteFilesCount = 0;
+
+                Object.values(files).forEach((file: IFileMetadata) => {
+                    if (file.authorParticipantId !== localParticipant?.id) {
+                        remoteFilesCount++;
+                    }
+                });
+
                 dispatch({
                     type: _FILE_LIST_RECEIVED,
-                    files
+                    files,
+                    remoteFilesCount
```

Comment:
Do we need to put this here? Can't the interested parties compue the value themselves?